src/sync_template.sh

```diff
@@ -52,10 +52,32 @@ echo "::endgroup::"
 echo "::group::Pull template"
 echo "::debug::create new branch from default branch with name ${NEW_BRANCH}"
 git checkout -b "${NEW_BRANCH}"
+
+if [ -s ${TEMPLATE_SYNC_IGNORE_FILE_NAME} ]
+then
+  echo "::debug::backup ignore file in case of incomming changes"
+  cp "${TEMPLATE_SYNC_IGNORE_FILE_NAME}" "${TEMPLATE_SYNC_IGNORE_FILE_NAME}.tmp"
+fi
+
 echo "::debug::pull changes from template"
 git pull "${SOURCE_REPO}" --allow-unrelated-histories --squash --strategy=recursive -X theirs
 echo "::endgroup::"
 
+if [ -s ${TEMPLATE_SYNC_IGNORE_FILE_NAME} ]
+then
+  echo "::group::restore ignore file"
+  # restore ignore file. the ignore file in the final repo always wins
+  mv -f "${TEMPLATE_SYNC_IGNORE_FILE_NAME}.tmp" "${TEMPLATE_SYNC_IGNORE_FILE_NAME}"
```

Comment:
The current approach definitely works in terms of the file.
@msiebeneicher I guess this will add a new commit to the file?

Maybe smth like `git checkout HEAD~1 -- ${TEMPLATE_SYNC_IGNORE_FILE_NAME}`  will reset it to current version so that no new commit will be done.

I did not test it (yet).

Reply:
Ok I have a possible solution now

```bash
git reset .templatesyncignore
git checkout -- .templatesyncignore
```

this should do the trick

Reply:
Nice catch! 10 times smarter than the other way! 👍 
Updated the code now with your solution.